src/Internal/ClusterDnsEndPointDiscoverer.php

```diff
@@ -90,8 +90,8 @@ public function __construct(
         $this->gossipTimeout = $gossipTimeout;
         $this->preferRandomNode = $preferRandomNode;
 
-        $this->client = new DefaultClient();
-        $this->client->setOption(Client::OP_TRANSFER_TIMEOUT, $gossipTimeout);
+        $this->client = new Client();
+        $this->client->addApplicationInterceptor(new SetRequestTimeout($gossipTimeout, $gossipTimeout, $gossipTimeout));
```

Comment:
It feels odd to use the timeout variable for 3 different timeouts here but I'm really unsure what else to do here...

Reply:
Yes, just leave it as is, that's fine.